content/doc/developer/publishing/documentation.adoc

```diff
@@ -0,0 +1,140 @@
+---
+title: Plugin Documentation
+layout: developersection
+references:
+- url: https://plugins.jenkins.io/
+  title: Plugin site
+- url: ../plugin-site/
+  title: Plugin site documentation
+- url: ../wiki-page/
+  title: Hosting plugin documentation on Jenkins Wiki
+---
+
+Documentation is an important part of any Jenkins plugin.
+It includes user documentation (plugin pages, changelogs, user guidelines, etc.) and the contributor documentation (how to contribute, developer guidelines, etc.).
+This section provides overview of these documentation types.
+
+== Plugin pages
+
+Plugin pages are hosted on the link:https://plugins.jenkins.io/[Plugin Site].
+These pages are generated automatically using the metadata from the latest plugin release and an external documentation page.
+External documentation can be retrieved from GitHub or from the https://wiki.jenkins.io[Jenkins Wiki].
+GitHub documentation support was introduced in September 2019 (https://groups.google.com/forum/#!topic/jenkinsci-dev/VSdfVMDIW-A[announcement]).
+
+For new plugins we recommend storing documentation in the GitHub repositories.
+It allows plugin maintainers to provide the same documentation from README pages and the Jenkins plugin site,
+and at the same time it allows using the Documentation-as-Code techniques when the documentation is a part of the
+repository and hence all common practices can be applied:
+
+* Pull requests and reviews
+* Creating documentation in parallel with features
+* Editing docs from GitHub Web UI, with preview support
+* Versioning, documentation for previous plugin versions can be easily accessed 
+
+=== Using GitHub as a source of documentation
+
+The plugin site can pull documentation from the root README pages or from other locations defined by the plugin URL (see below).
+Multiple formats are supported: GitHub Markdown, Asciidoc or raw text.
+
+To publish the plugin documentation on GitHub...
+
+. Create a README page and put the plugin documentation there.
+  This page will become a landing page for the link:https://plugins.jenkins.io/[Plugin site].
+** More documentation pages can be introduced inside the repository and
+linked from the README, the plugin site will display both absolute and
+relative links
+** Images from pages will be displayed by the plugin site as well
+. Modify your project URL to point to the GitHub repository, e.g. `http://github.com/jenkinsci/your-plugin`.
+  See the guidelines for Maven and Gradle below.
+. Release the new plugin version.
+  After several hours, the link:https://plugins.jenkins.io/[Plugin site] with start using it as a documentation source.
```

Comment:
```suggestion
  After several hours, the link:https://plugins.jenkins.io/[Plugin site] should display the new documentation
```

better now?

Reply:
How about this?

```suggestion
  Once the link:https://plugins.jenkins.io/[plugin site] picks up the release (which can take up to a few hours) it will also display the documentation from GitHub.
```

(Assuming we invalidate plugin documentation caches when the URL changes.)